### PR TITLE
Bump Scheduler to use Kafka Consumer v0.4

### PR DESCRIPTION
**Pull request by jppierri**
(PD internal linked JIRA ticket: CORE-1067)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -67,7 +67,7 @@ lazy val common = (project in file("common")).
       "org.json4s"   %% "json4s-jackson" % "3.3.0",
       "org.slf4j" % "slf4j-api" % "1.7.13",
       "org.slf4j" % "jul-to-slf4j" % "1.7.13",
-      "org.apache.kafka" % "kafka-clients" % "0.9.0.1",
+      "org.apache.kafka" % "kafka-clients" % "0.10.1.0",
```
Comment (cdegroot) [inline build.sbt]: Shouldn't this be pulled in through a transitive dependency on the kafka-consumer lib?

File: version.sbt [hunk 2]
```diff
@@ -1 +1 @@
-version in ThisBuild := "8.0.0"
+version in ThisBuild := "8.1.0"
```
Comment (cdegroot) [inline version.sbt]: Technically maybe a minor version. But given that we do a major bump in Kafka deps, maybe worth a major bump as well?

File: build.sbt (hunk 1 above)
Comment (jppierri) [inline build.sbt]: Sure, let's just remove this explicit dependency entirely.

File: version.sbt (hunk 2 above)
Comment (jppierri) [inline version.sbt]: Sure.

**Pull request merged**